airflow/providers/common/sql/hooks/sql.py

```diff
@@ -516,3 +517,40 @@ def test_connection(self):
             message = str(e)
 
         return status, message
+
+    def get_database_info(self, connection):
+        from airflow.providers.openlineage.sqlparser import DatabaseInfo
+
+        return DatabaseInfo(
+            scheme=self.get_database_dialect(connection), authority=self._get_authority(connection)
+        )
+
+    def get_database_dialect(self, connection):
+        """Method used for SQL parsing. Naively tries to use Connection's conn_type"""
+        return connection.conn_type
+
+    def _get_authority(self, connection):
+        """Returns authority part (without user info) of OpenLineage namespace URI."""
+        if connection.port and connection.host:
+            authority = f"{connection.host}:{connection.port}"
+        else:
+            parsed = urlparse(connection.get_uri())
+            authority = f"{parsed.hostname}:{parsed.port}"
+        return authority
+
+    def get_default_schema(self):
+        """
+        Returns default schema specific to database.
+        See: :class:`~providers.openlineage.utils.sqlparser.SQLParser`
+        """
+        return self.__schema or "public"
+
+    def get_database_specific_lineage(self, task_instance):
```

Comment:
isn't there specific lineage in case of snowflake - eg. query run time

Reply:
For now I'm trying to reflect what's currently sent from `openlineage-airflow` package.
I might add some more information but I'd really appreciate what would be useful and availabe to retrieve.

Reply:
On the other hand - https://docs.snowflake.com/en/sql-reference/account-usage/query_history#usage-notes: latency for `QUERY_HISTORY` is up to 45 minutes so it looks like it's not sustainable to _always_ get_ query run time